packages/cli/src/commands/deploy/action.ts

```diff
@@ -24,6 +24,8 @@ function isProxyKind(kind: Options['kind']): boolean {
     case 'upgradeable':
     case 'minimal':
       return true;
+    default:
+      throw new Error(`Unknown proxy kind: ${kind}`);
```

Comment:
@frangio had brought up a way to handle this so the compiler itself complains (creating a new exception type relying on type never to ensure the exception class is never instantiated). Maybe we want to do sth like that in a subsequent PR?

Reply:
Not sure If I follow the idea correct. Is there an example of doing that?

Reply:
The problem with this `default:` is that it nullifies the compiler's checks. There won't be a type error in this function if we add a a new proxy kind because there's already a default case to handle it.

It's explained here: https://2ality.com/2020/02/typescript-exhaustiveness-checks-via-exceptions.html